### PR TITLE
Feature/cucumber smoke tests

### PR DESCRIPTION
**Pull request by mikeSmith08**
Refactored the step js files out so that they can be used by both the acceptance tests and the smoke tests, plus a fair bit of tidying around the use of environment variables and config files.

One nice bonus here is that we can now remove the use of Mocha.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
...
-    "test:lint": "standard --env mocha",
+    "test:lint": "standard",
     "test:unit": "nyc tape -r './src/test/unit/setup' 'src/@(web|config)/**/*.test.js'",
-    "test:acceptance": "mkdir -p build/reports && cucumber-js src/test/acceptance/features/ -f json:build/reports/cucumber-report.json",
-    "test:acceptance:report": "node src/test/cucumber-report.js",
-    "test:smoke": "mocha 'src/test/smoke/test-suite.js'",
+    "test:acceptance": "mkdir -p build/reports && cross-env TESTS=acceptance cucumber-js src/test/acceptance/features/ --require 'src/test/common/steps/*.js' --require 'src/test/acceptance/config' -f json:build/reports/acceptance-report.json",
+    "test:acceptance:report": "node src/test/acceptance-report.js",
+    "test:smoke": "npm-run-all -s test:smoke:test test:smoke:report",
+    "test:smoke:test": "mkdir -p build/reports && cross-env TESTS=smoke cucumber-js src/test/smoke/features/ --require 'src/test/common/steps/*.js' --require 'src/test/smoke/config' -f json:build/reports/smoke-report.json",
```
Comment (gethyn1) [inline package.json]: Some of these scripts are getting quite long, should we consider moving to separate files? Or maybe it doesn't matter?

File: package.json [hunk 2]
```diff
@@ -29,9 +31,9 @@
     "chai": "^4.2.0",
     "chromedriver": "^2.45.0",
     "codecov": "^3.1.0",
+    "cross-env": "^5.2.0",
     "cucumber": "^5.1.0",
     "cucumber-html-reporter": "^4.0.4",
-    "mocha": "^5.2.0",
```
Comment (gethyn1) [inline package.json]: YAY!!

File: package.json [hunk 3]
```diff
@@ -29,9 +31,9 @@
     "chai": "^4.2.0",
     "chromedriver": "^2.45.0",
     "codecov": "^3.1.0",
+    "cross-env": "^5.2.0",
```
[the remark this answers is not in the document]
Comment (mikeSmith08) [reply package.json]: cross-env is only used for acceptance and smoke tests, I think dotenv is used for more that that, do they have the same scope? Which scope do you think is best, if they're both only used for tests, then is devDependencies the correct place?

File: src/test/common/steps/pages.js [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+const EnterName = require('../page/enter-name')
+const Overview = require('../page/overview')
+const Confirmation = require('../page/confirmation')
+const DriverManager = require('../driver/driver-manager')
+const { ACCEPTANCE_BASE_URL } = require('../../acceptance/config/constants')
+const { SMOKE_BASE_URL } = require('../../smoke/config/environment')
 
 const driverManager = new DriverManager()
 
+const TESTS = process.env.TESTS
```
[the remark this answers is not in the document]
Comment (mikeSmith08) [reply src/test/common/steps/pages.js]: Ok, I'll do that.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (gethyn1) [reply package.json]: We should only be using `dotenv` locally so maybe they should both be in `devDependencies`

File: package.json (hunk 1 above)
Comment (mikeSmith08) [reply package.json]: I think that's my thought, so let's leave it for now.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikeSmith08) [reply package.json]: Ok, I'll do that, devDependencies it is.

File: src/test/common/steps/pages.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mikeSmith08) [reply src/test/common/steps/pages.js]: Done.

**Pull request merged**